src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -268,7 +334,10 @@ private SemanticRange DataToSemanticRange(int lineDelta, int charDelta, int leng
                 ResultId = resultId.ToString()
             };
 
-            _semanticTokensCache.Set(resultId.ToString(), (versionStamp, data));
+            if (complete)
+            {
+                _semanticTokensCache.Set(resultId.ToString(), (versionStamp, data));
```

Comment:
Should we be updating the cache in the caller instead of here? Feels odd to conditionally set the cache in a convert method anyhow.

Reply:
I'ma level with you, I don't remember what caused me to think I needed this condition, I can't think of any scenario where There would be a null `SemanticRange`. I'll also move it up since you're right that a convert method is a weird spot.